apbs/create-secret-and-configmap-during-provision.md

```diff
@@ -0,0 +1,44 @@
+# Abstract
+
+This document outlines a proposal to create an additional configmap for services accessible via a mobile client, this configmap would compliment the secret we currently create during the provision of each service via an APB 
+
+## Terms
+
+- Mobile Client (a mobile device in the public domain running a developers application)
+- APB (ansible playbook bundle)
+
+## Problem Description
+
+Our current approach is to create a secret during the provision of a new mobile service. This secret contains all of the information required to allow the UI to show the service and how to access the service, it also includes any information needed to create the mobile client configuration. This means we expose too much information and also need to filter out any sensitive information (usernames password etc) when constructing our mobile configuration to avoid it being exposed to the client and ending up in the public domain. As each service is different, it often requires a custom piece of code to filter out the fields that we don't want or need exposed. 
+
+## Proposed Solution
+
+When provisioning a new service instance, we should create two configuration resources:
+
+1) A secret containing private information such as usernames and passwords or anything that we do not want to be exposed to the mobile client.
+2) A configmap containing all of the required information needed for a mobile client to access the service.
+
+Note it may be neccessary to duplicate some information across each of these resources (ie the URI of the service for example).
+
+Each of these resources should have at least the following information and labels:
+
+- Labels: 
+    - **mobile:enabled**. Will give a mechanism to filter these resources from other resources of the same type from the UI and CLI
```

Comment:
`mobile:true` might be better as it could be parsed into a boolean easier if needed. Also, is it something that can be 'disabled'? A boolean says it is or isn't something rather than something that's enabled or disabled.

Reply:
the label is just an identifier a way of filtering. Also labels can only be strings so cant really use true false unless they are parsed in the client and then there is no difference to using enabled

Reply:
OK, only strings makes sense then. JSON.parse or equivalent on a client would parse "true" to a boolean

Reply:
Really? That seems bad. In Javascript maybe it does that. Pretty sure that wouldn't work in Go though

Reply:
Yeah, e.g
```
typeof JSON.parse('true')
'boolean'
```
anyways, :+1: on this